website/examples/date-picker-dialog.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import { DayPicker } from 'react-day-picker';
+import { usePopper } from 'react-popper';
+
+import { format, isValid, parse } from 'date-fns';
+import FocusTrap from 'focus-trap-react';
+
+export default function Example() {
+  const [selected, setSelected] = React.useState<Date>();
+  const [inputValue, setInputValue] = React.useState<string>('');
+  const [isOpen, isOpenSet] = React.useState(false);
+
+  const refEl = React.useRef<HTMLDivElement>(null);
+  const buttonRef = React.useRef<HTMLButtonElement>(null);
+  const [popperEl, popperElSet] =
+    React.useState<HTMLDivElement | null>(null);
+
+  const closePopper = () => {
+    isOpenSet(false);
+    buttonRef?.current?.focus();
+  };
+
+  const { styles, attributes } = usePopper(
+    refEl.current,
+    popperEl,
+    {
+      placement: 'bottom-start'
+    }
+  );
+
+  const dateFormat = 'P';
+  const formatDate = (date: Date) => format(date, dateFormat);
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const text = event.currentTarget.value;
+    setInputValue(text);
+    const date = parse(text, dateFormat, new Date());
+    if (isValid(date)) {
+      setSelected(date);
+    } else {
+      setSelected(undefined);
+    }
+  };
+
+  const handleDaySelected = (date: Date | undefined) => {
+    setSelected(date);
+    if (date) {
+      setInputValue(formatDate(date));
+      closePopper();
+    } else {
+      setInputValue('');
+    }
+  };
+
+  return (
+    <div style={{ position: 'relative' }}>
+      <div
+        ref={refEl}
+        style={{
+          padding: '4px 0',
+          display: 'flex',
+          alignItems: 'center'
+        }}
+      >
+        Date
+        <input
+          type="text"
+          style={{
+            marginLeft: '4px',
+            padding: '6px 4px',
+            lineHeight: '24px',
+            fontSize: '16px',
+            height: '40px'
+          }}
+          placeholder="mm/dd/yyyy"
```

Comment:
```suggestion
          placeholder={format(new Date(), 'P')}
```

Reply:
yes, but then that changes the placeholder to today's day, which is too arbitrary in my mind for a placeholder -- not that pattern letters are much better...

Reply:
That's fine for me this placeholder to… Yet, if you change the formatter in `formatDate` few lines above, this should be updated anyway.  I think i prefer to move these out of the function:

```ts
const formatString = 'P';
const placeholder = 'mm/dd/yyyy';

export default function Example() {
```

remove the custom `formatDate`, and always use `format` from date-fns.

Reply:
Hmm…yeah…hmm…mm/dd/yyyy is not very well localized. Maybe ‘P’ can be translated into some format string that is localized, but I guess I’ll have to think about it some more

Reply:
I couldn't find any way to get a localized pattern -- I left the en-US-style pattern, with a comment to readers to localize as they see fit...